x-pack/plugins/alerting/server/lib/monitoring.test.ts

```diff
@@ -70,23 +79,56 @@ describe('getExecutionDurationPercentiles', () => {
 
   it('Returns empty object when given all undefined durations', () => {
     // remove all duration fields
-    const nullDurationHistory = mockHistory.map((history) => ({
+    const nullDurationHistory: RuleMonitoringHistory[] = mockHistory.map((history) => ({
+      timestamp: history.timestamp,
       success: history.success,
     }));
 
-    const newMockRuleMonitoring = {
+    const newMockRuleMonitoring: RuleMonitoring = {
       ...mockRuleMonitoring,
       run: {
         ...mockRuleMonitoring.run,
         history: nullDurationHistory,
       },
-    } as RuleMonitoring;
+    };
 
     const percentiles = getExecutionDurationPercentiles(newMockRuleMonitoring.run.history);
     expect(Object.keys(percentiles).length).toEqual(0);
   });
 });
 
+describe('resetMonitoringLastRun', () => {
+  it('resets last run metrics to the initial default value', () => {
+    const result = resetMonitoringLastRun(mockRuleMonitoring);
+    expect(result.run.last_run.metrics).toEqual({
+      duration: 0,
+      total_search_duration_ms: null,
+      total_indexing_duration_ms: null,
+      total_alerts_detected: null,
+      total_alerts_created: null,
+      gap_duration_s: null,
+    });
+  });
+
+  it('preserves last run timestamp', () => {
+    const result = resetMonitoringLastRun(mockRuleMonitoring);
+    expect(result.run.last_run.timestamp).toEqual(mockRuleMonitoring.run.last_run.timestamp);
```

Comment:
If `resetMonitoringLastRun()` modifies the object itself so this test will pass while the function's implementation will be incorrect. We can't guarantee it doesn't happen in the future. It'd be much better to compare with a constant value like this

```
expect(result.run.last_run.timestamp).toEqual(MOCK_TIMESTAMP);
```

Reply:
Nice catch!